pyrit/prompt_target/openai/openai_response_target.py

```diff
@@ -0,0 +1,342 @@
+            # "type" and "text" at a deeper level, they are just a single content string.
+            # The "system" role is mapped to "developer" in the OpenAI Response API.
+            first_piece = prompt_request_pieces[0]
+            if first_piece.role == "system":
+                if len(prompt_request_pieces) > 1:
+                    raise ValueError(
+                        "System messages should only have a single request piece. "
+                        "Multiple system messages are not supported."
+                    )
+                full_request.append(
+                    ChatMessage(role="developer", content=first_piece.converted_value).model_dump(exclude_none=True)
+                )
+                continue
+
+            content = []
+            for prompt_request_piece in prompt_request_pieces:
+                role = prompt_request_piece.role
+                if prompt_request_piece.converted_value_data_type == "text":
+                    if role == "user":
+                        type = "input_text"
+                    else:
+                        type = "output_text"
+                    entry = {"type": type, "text": prompt_request_piece.converted_value}
+                    content.append(entry)
+                elif prompt_request_piece.converted_value_data_type == "image_path":
+                    data_base64_encoded_url = await convert_local_image_to_data_url(
+                        prompt_request_piece.converted_value
+                    )
+                    image_url_entry = {"url": data_base64_encoded_url}
+                    entry = {"type": "input_image", "image_url": image_url_entry}  # type: ignore
+                    content.append(entry)
+                elif prompt_request_piece.converted_value_data_type == "reasoning":
+                    # reasoning summaries are not passed back to the target
+                    pass
+                else:
+                    raise ValueError(
+                        f"Multimodal data type {prompt_request_piece.converted_value_data_type} is not yet supported."
+                    )
+
+            full_request.append(
+                ChatMessageListDictContent(role=role, content=content).model_dump(exclude_none=True)  # type: ignore
+            )
+
+        return full_request
+
+    async def _construct_request_body(
+        self, conversation: MutableSequence[PromptRequestResponse], is_json_response: bool
+    ) -> dict:
+        input = await self._build_input_for_multi_modal_async(conversation)
+
+        body_parameters = {
+            "model": self._model_name,
+            "max_output_tokens": self._max_output_tokens,
+            "temperature": self._temperature,
+            "top_p": self._top_p,
+            "stream": False,
+            "input": input,
+            # json_schema not yet supported by PyRIT
+            "response_format": {"type": "json_object"} if is_json_response else None,
```

Comment:
Is `response_format` supported by the Responses API? Took a look at the API Ref and I might be wrong, but it seems like it might be the below?
```suggestion
            "text": {"format": {"type": "json_object"}} if is_json_response else None,
```

Reply:
https://platform.openai.com/docs/api-reference/chat/create#chat-create-response_format

Where did you find yours 👀

Reply:
Isn't that for Chat Completions, not Responses API? https://platform.openai.com/docs/api-reference/responses/create#responses-create-text

Reply:
Thanks for explaining offline. Good callout!